tests/lib/prepare-project.sh

```diff
@@ -44,62 +44,71 @@ build_deb(){
     cp ../*.deb "$GOHOME"
 }
 
-fedora_build_rpm() {
+build_rpm() {
+    distro=$(echo "$SPREAD_SYSTEM" | awk '{split($0,a,"-");print a[1]}')
     release=$(echo "$SPREAD_SYSTEM" | awk '{split($0,a,"-");print a[2]}')
     arch=x86_64
-
-    base_version="$(head -1 debian/changelog | awk -F'[()]' '{print $2}')"
-    version="1337.$base_version"
-    sed -i -e "s/^Version:.*$/Version: $version/g" packaging/fedora-$release/snapd.spec
-
-    mkdir -p /tmp/pkg/snapd-$version
-    cp -rav * /tmp/pkg/snapd-$version/
-
-    mkdir -p $HOME/rpmbuild/SOURCES
-    (cd /tmp/pkg; tar czf $HOME/rpmbuild/SOURCES/snapd-$version.tar.gz snapd-$version --exclude=vendor/)
-
-    cp packaging/fedora-$release/* $HOME/rpmbuild/SOURCES/
-
-    rpmbuild -bs packaging/fedora-$release/snapd.spec
-    mock -v /root/rpmbuild/SRPMS/snapd-$version-*.src.rpm
-    cp /var/lib/mock/fedora-$release-$arch/result/*.rpm $GOPATH
-    rm $GOPATH/*.src.rpm
-}
-
-opensuse_build_rpm() {
-    release=$(echo "$SPREAD_SYSTEM" | awk '{split($0,a,"-");print a[2]}')
-    arch=x86_64
-
     base_version="$(head -1 debian/changelog | awk -F'[()]' '{print $2}')"
     version="1337.$base_version"
-    sed -i -e "s/^Version:.*$/Version: $version/g" packaging/opensuse-$release/snapd.spec
+    packaging_path=packaging/$distro-$release
+    archive_name=snapd-$version.tar.gz
+    archive_compression=z
+    extra_tar_args=
+    rpm_dir=
 
+    case "$SPREAD_SYSTEM" in
+        fedora-*)
+            extra_tar_args="$extra_tar_args --exclude=vendor/"
+            rpm_dir=$HOME/rpmbuild
+            ;;
+        opensuse-*)
+            rpm_dir=/usr/src/packages
```

Comment:
You can set `rpmbuild` to use the same path in both places.

just write the following to `~/.rpmmacros`:

```spec
%_topdir %(echo $HOME)/rpmbuild
```